src/bot/root_pythia_bot.py

```diff
@@ -50,6 +56,9 @@ def craft_intents():
 
 
 ########### Setup bot events response ###############
+@BOT.check
+def is_my_channel(ctx):
```

Comment:
Is it ran by the bot without explicitely asking it to ?

Reply:
Yes, the decorator `@BOT.check` registers it as a [global check](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=commands%20bot#discord.ext.commands.Bot.check) run for each command, if it doesn't pass it raises [CheckFailure later caught by `on_command_error`](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=commands%20bot#discord.ext.commands.check)
> These checks should be predicates that take in a single parameter taking a [Context](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=commands%20bot#discord.ext.commands.Context). If the check returns a False-like value then during invocation a [CheckFailure](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=commands%20bot#discord.ext.commands.CheckFailure) exception is raised and sent to the [on_command_error()](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=commands%20bot#discord.discord.ext.commands.on_command_error) event.

But more practically you can test it:
```
git clone https://github.com/iScsc/RootPythia.git
## fill in the .env.dev with values I can give you
./run.sh dev

## the following commands are in the conatiner:
python src/main.py
# it started the bot, just try to trigger the bot from a wrong channel and check your logs
```

Reply:
would you like me to add a regression unit test?

Reply:
Ok seems fine to me then !
Thanks for your answer